File: 24-1/HONG/24-1-22/9046.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {

    int T;
    cin >> T;
    cin.ignore();

    for (int t = 0; t < T; t++) {
        string line;
        int cnt[26] = {0};

        getline(cin, line);
        for (int i = 0; line[i] != 0; i++) {
            if (line[i] == ' ') {
                continue;
            }
            int order = line[i] - 'a';
            cnt[order] += 1;
        }

        int max_i = 0;
        for (int i = 1; i < 26; i++) {
            if (cnt[i] > cnt[max_i]) {
                max_i = i;
            }
        }

        int cnt_same = 0;
        for (int i = 0; i < 26; i++) {
            if (cnt[i] == cnt[max_i]) {
                cnt_same += 1;
            }
        }

        if (cnt_same == 1) {
            cout << char('a' + max_i) << "\n";
        } else {
            cout << "?\n";
        }
    }

    return 0;
}

File: 24-1/HONG/24-1-22/9046.py
T = int(input())

for t in range(T):
  line = input()
  abc = [0] * 26

  for ch in line:
    if ch == ' ':
      continue
    order = ord(ch) - ord('a')
    abc[order] += 1
  
  max_i = 0
  for i in range(1, 26):
    if abc[i] > abc[max_i]:
      max_i = i
  
  same_cnt = 0
  for i in range(26):
    if abc[max_i] == abc[i]:
      same_cnt += 1
  
  if same_cnt == 1:
    print(chr(max_i + ord('a')))
  else:
    print('?')